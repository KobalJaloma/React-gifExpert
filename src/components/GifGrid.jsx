
import { useFetchGifts } from "../hooks/useFetchGifts";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
    
    const { images, isLoading} = useFetchGifts( category );

    return (
    <div>
        <h3>{category}</h3>
        {
            isLoading && (<h2>Loading...</h2>) 
        }
        <h2></h2>

        <div className="card-grid">
            {
                images.map( (image) => {
                    return(
                        <GifItem 
                            key={image.id} 
                            {...image}
                        />
                    )
                })
            }
        </div>
    </div>
  )
}

// {...image} CARGAR PROPIEDADES DEL ELEMENTO DESESTRUCTURADA