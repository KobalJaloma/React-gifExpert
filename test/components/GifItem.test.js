import { GifItem } from '../../src/components/GifItem';
import { screen, render } from "@testing-library/react";

describe('Testing GifItem', () => { 
    
    const title = "RandomTitle"
    const url = "https://algo.com/"
    
    test('match in the Snapshoot', () => { 
            

        const { container } = render( <GifItem key={1} title={title} url={url}/> )
        expect( container ).toMatchSnapshot();

    })

    test('Show the image with URL and ALT correspondent', () => { 
        
        render(<GifItem title={title} url={url}/>)
        const {alt, src} = screen.getByRole('img');

        expect(alt).toBe(title);
        expect(src).toBe(url)
        
     })



 })