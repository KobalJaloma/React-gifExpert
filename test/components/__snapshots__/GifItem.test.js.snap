// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem match in the Snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="RandomTitle"
      src="https://algo.com/"
    />
    <p>
      RandomTitle
    </p>
  </div>
</div>
`;
